lib/nsq/logger.ex

```diff
@@ -0,0 +1,23 @@
+defmodule NSQ.Logger do
+  require Logger
+
+  defdelegate debug(message), to: Logger
+  defdelegate info(message), to: Logger
+  defdelegate error(message), to: Logger
+
+  def warn(message) do
+    case Version.compare(System.version(), "1.11.0") do
+      :lt -> Logger.warn(message)
+      _ -> Logger.warning(message)
+    end
+  end
+
+  def configure(opts) do
+    case {opts, Version.compare(System.version(), "1.11.0")} do
+      {[level: :warn], :gt} ->
+        Logger.configure(Keyword.merge(opts, [level: :warning]))
+      _ ->
+        Logger.configure(opts)
+    end
+  end
```

Comment:
As I understand this is not strictly necessary, but I am ok with keeping it.
:warn is still supported in the opts: https://github.com/elixir-lang/elixir/blob/81a12b777411e5fb3dc48cb314771c03202e0679/lib/logger/lib/logger.ex#L1164